src/backend/tcop/zenith_wal_redo.c

```diff
@@ -565,6 +571,7 @@ PushPage(StringInfo input_message)
 	content = pq_getmsgbytes(input_message, BLCKSZ);
 
 	buf = ReadBufferWithoutRelcache(rnode, forknum, blknum, RBM_ZERO_AND_LOCK, NULL);
+	Assert(!BufferIsInvalid(buf));
```

Comment:
This assertion seems excessive. ReadBufferWithoutRelcache with RBM_ZERO_AND_LOCK cannot return an InvalidBuffer, and it doesn't seem like all callers should need to assert that. Also, BufferGetPage below has an assertion like that already.

```suggestion
```